spec/system/device_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+RSpec.describe 'Log in', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  describe 'loads' do
+    before :each do
+      visit new_user_session_path
+    end
+
+    it 'the username and password inputs and the "Submit" button.' do
+      expect(page).to have_content 'Email'

```

Comment:
✅ Has featured test for login page 👍🏻. This is a good trial on writing first test. You can also use the suggestion below and find away to refactor your test
```ruby
describe 'the signin process', type: :feature do
     before :each do
         User.create(email: 'user1@gmail.com', password: 'password', name: 'User1')
     end

     it 'signs @user in' do
         visit '/users/sign_in'
         fill_in 'Email', with: 'user1@gmail.com'
         fill_in 'Password', with: 'password'
         click_button 'Log in'
         expect(current_path).to eq(root_path)
         expect(page).to have_text('Signed in successfully.')
     end

     it 'does not sign in with wrong info' do
         visit '/users/sign_in'
         fill_in 'Email', with: 'user1@gmail.com'
         fill_in 'Password', with: 'mypassword'
         click_button 'Log in'
         expect(page).to have_text('Invalid Email or password.')
     end
 end
```